RealmSwift/Map.swift

```diff
@@ -0,0 +1,600 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import Realm
+import Realm.Private
+
+/// :nodoc:
+public protocol _MapKey: Hashable {
+    static var _rlmType: RLMPropertyType { get }
+}
+extension String: _MapKey { }
+
+/**
+ Map is a key-value storage container used to store supported Realm types.
+ 
+ Map is a generic type that is parameterized on the type it stores. This can be either an Object
+ subclass or one of the following types: Bool, Int, Int8, Int16, Int32, Int64, Float, Double,
+ String, Data, Date, Decimal128, and ObjectId (and their optional versions)
```

Comment:
This should note that their optional versions are only supported in non-synchronized Realms.